docs/source/howto/plugin_codes.rst

```diff
@@ -223,104 +250,205 @@ The Topics section on :ref:`defining processes <topics:processes:usage:defining>
+ * You can submit calcualtions to the AiiDA daemon without having to worry about how to make sure that your calculation and parser plugins can be imported from the daemon's python environment.
 
-Performing a dry-run
---------------------
+Registering entry points
+------------------------
 
-Additionally, you might want to check and verify your inputs before actually running or submitting a calculation.
-You can do so by specifying to use a ``dry_run``, which will create all the input files in a local directory (``submit_test/[date]-0000[x]``) so you can inspect them before actually launching the calculation:
+Registering entry points is simple:
 
-.. code-block:: python
+ * Write a minimalistic ``setup.py`` script:
 
-    calculation_builder.metadata.dry_run = True
-    calculation_builder.metadata.store_provenance = False
-    run(calculation_builder)
+    .. code-block:: python
 
-.. todo::
+        from setuptools import setup
+
+        setup(
+            name='aiida-add',
+            packages=['aiida_add'],
+            entry_points={
+                'aiida.calculations': ["add = aiida_add.calcjob:ArithmeticAddCalculation"],
+                'aiida.parsers': ["add = aiida_add.parser:ArithmeticAddParser"],
+            }
+        )
+
+ * Move your two scripts into a subfolder ``aiida_add``:
+
+   .. code-block:: console
+
+      mkdir aiida_add
+      mv calcjob.py parser.py aiida_add/
+
+ * Install your new ``aiida-add`` plugin package:
+
+   .. code-block:: console
+
+      pip install -e .
+      reentry scan
+
+
+After this, you should already see your calculation plugin listed:
+
+   .. code-block:: console
+
+      $ verdi plugin list aiida.calculations
+      $ verdi plugin list aiida.calculations add
+
+
+Launching your calculation now becomes easier:
+
+ * In ``calcjob.py``, change the default  ``parser_name`` to ``'add'``
+
+ * Simplify your ``launch.py`` script:
+
+    .. code-block:: python
+
+        from aiida import orm, engine
 
-    .. _how-to:plugin-codes:caching:
+        # Setting up inputs
+        computer = orm.load_computer('localhost')
+        code = orm.Code(remote_computer_exec=[computer, '/bin/bash'], input_plugin_name='add')
+        builder = code.get_builder()
+        builder.x = Int(4)
+        builder.y = Int(5)
+        builder.metadata.options.withmpi = False
+        builder.metadata.options.resources = {
+            'num_machines': 1,
+            'num_mpiprocs_per_machine': 1,
+        }
 
-    title: Using caching to save computational resources
+        # Running the calculation & parsing results
+        output_dict, node = engine.run_get_node(builder)
+        print("Parsing completed. Result: {}".format(int(output_dict['sum'])))
 
-    `#3988`_
+    .. note::
 
+        ``output_dict`` is a dictionary containing all the output nodes keyed after their label.
+        In this case: "remote_folder", "retrieved" and "sum".
```

Comment:
Why not put this after the first launch script?

Reply:
There were already other comments made there, so I decided to delay it.
But you are certainly correct that it would fit there as well.